x-pack/plugins/serverless_search/public/application/components/connectors/connector_config/connector_link.tsx

```diff
@@ -87,11 +87,28 @@ export const ConnectorLinkElasticsearch: React.FC<ConnectorLinkElasticsearchProp
           </EuiFlexItem>
         </EuiFlexGroup>
         <EuiSpacer />
-        <EuiFlexItem>
-          <ConnectionDetails connectorId={connectorId} serviceType={serviceType} status={status} />
-        </EuiFlexItem>
+        {Boolean(serviceType) ? (
+          <EuiFlexItem>
+            <ConnectionDetails
+              connectorId={connectorId}
+              serviceType={serviceType}
+              status={status}
+            />
+          </EuiFlexItem>
+        ) : (
+          <EuiFlexItem>
+            <EuiCallOut
+              title={i18n.translate('xpack.serverlessSearch.connectors.pleaseSelectServiceType', {
+                defaultMessage: 'Please select a connector type.',
+              })}
+              color="warning"
+              iconType="iInCircle"
+            />
+          </EuiFlexItem>
+        )}
         <EuiSpacer />
-        {status === ConnectorStatus.CREATED ? (
+        {Boolean(serviceType) &&
+        (status === ConnectorStatus.CREATED || status === ConnectorStatus.NEEDS_CONFIGURATION) ? (
```

Comment:
This change confuses me a little. Are we sure this is correct? We've had issues where we get people stuck at a step in this logic.

Reply:
After the service type is selected, the connector status becomes `needs_configuration` despite not being connected yet. Because of that, the `Waiting for connection` card disappears even though the Connector hasn't checked in yet. If you try it on prod now you'll see that the card disappears when it shouldn't.

I think the `Boolean(serviceType)` could be removed, I just didn't want two of the yellow cards to appear at the bottom at the same time ("select connector type" _and_ "waiting for connection")

Reply:
Here's a gif of how it works right now on `main`. Notice that the "waiting for connection" card disappears despite the connector not being connected yet (all I've done is selected a service type).

![out](https://github.com/user-attachments/assets/e752a9c4-20d4-40ff-8d24-6d9130993b99)